src/engine/source/hlp/src/parsers/date.cpp

```diff
@@ -155,7 +158,75 @@ std::string formatDateFromSample(const std::string& dateSample, const std::strin
 }
 } // namespace
 
-namespace hlp::parsers
+namespace hlp
+{
+
+namespace
+{
+/**
+ * @brief Load the timezone database and check if the version is the same as the provided.
+ *
+ * @param version The version to check
+ * @return true if the database is loaded and the version is the same as the provided
+ */
+bool loadTimeZoneDB(const std::string& version)
```

Comment:
nitpeak: Shouldn't you check that this is not empty? The remove_version method does not ensure this.

Reply:
It is not really important whether it is empty or not. If you cannot determine the database, depending on the error it may be empty or "unknown". This works like this in the date library with the macro `AUTO_DOWNLOAD=1`. That is why it is compared that the versions are not the same, without making comparisons of lesser equals (`return version == db.version`)